be/src/exprs/agg/percentile_cont.h

```diff
@@ -92,18 +208,29 @@ class PercentileContDiscAggregateFunction
         Bytes& bytes = column->get_bytes();
         size_t old_size = bytes.size();
         size_t items_size = this->data(state).items.size();
+        size_t grid_items_size = 0;
+        for (const auto& vec : this->data(state).grid) {
+            grid_items_size += vec.size() - 2;
+        }
+        size_t total_items_size = items_size + grid_items_size;
 
         // should serialize: rate_size, vector_size, all vector element.
-        size_t new_size = old_size + sizeof(double) + sizeof(size_t) + items_size * sizeof(InputCppType);
+        size_t new_size = old_size + sizeof(double) + sizeof(size_t) + total_items_size * sizeof(InputCppType);
         bytes.resize(new_size);
 
         memcpy(bytes.data() + old_size, &(this->data(state).rate), sizeof(double));
-        memcpy(bytes.data() + old_size + sizeof(double), &items_size, sizeof(size_t));
+        memcpy(bytes.data() + old_size + sizeof(double), &total_items_size, sizeof(size_t));
         memcpy(bytes.data() + old_size + sizeof(double) + sizeof(size_t), this->data(state).items.data(),
                items_size * sizeof(InputCppType));
-        pdqsort(reinterpret_cast<InputCppType*>(bytes.data() + old_size + sizeof(double) + sizeof(size_t)),
-                reinterpret_cast<InputCppType*>(bytes.data() + old_size + sizeof(double) + sizeof(size_t) +
-                                                items_size * sizeof(InputCppType)));
+        size_t current_pos = old_size + sizeof(double) + sizeof(size_t) + items_size * sizeof(InputCppType);
+        for (const auto& vec : this->data(state).grid) {
+            memcpy(bytes.data() + current_pos, vec.data() + 1, (vec.size() - 2) * sizeof(InputCppType));
+            current_pos += (vec.size() - 2) * sizeof(InputCppType);
+        }
+
+        std::sort(reinterpret_cast<InputCppType*>(bytes.data() + old_size + sizeof(double) + sizeof(size_t)),
```

Comment:
why not use `pdqsort` ? it's faster than `std::sort`

Reply:
Already modified, thanks a lot